components/script/dom/webglrenderingcontext.rs

```diff
@@ -3805,48 +3805,50 @@ impl UniformSetterType {
     }
 }
 
-#[derive(Default, JSTraceable, MallocSizeOf)]
+#[derive(JSTraceable, MallocSizeOf)]
 #[must_root]
-pub struct BoundAttribBuffers {
-    elements: DomRefCell<FnvHashMap<u32, (bool, Option<Dom<WebGLBuffer>>)>>,
+pub struct VertexAttribs {
+    attribs: DomRefCell<Box<[(bool, Option<Dom<WebGLBuffer>>)]>>,
 }
 
-impl BoundAttribBuffers {
+impl VertexAttribs {
+    pub fn new(max: u32) -> Self {
+        Self { attribs: DomRefCell::new(vec![Default::default(); max as usize].into()) }
```

Comment:
How big can `max_vertex_attribs` be vs. what it's used? May it be worth it to grow it lazily or something?